modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/DefaultExtensionsClient.java

```diff
@@ -0,0 +1,149 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.client;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.module.extension.internal.util.MuleExtensionUtils.getInitialiserEvent;
+import static reactor.core.publisher.Flux.from;
+import static reactor.core.publisher.Mono.just;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.lifecycle.Initialisable;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.message.Attributes;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.policy.DefaultPolicyManager;
+import org.mule.runtime.core.policy.PolicyManager;
+import org.mule.runtime.extension.api.ExtensionManager;
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.client.OperationParameters;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.module.extension.internal.manager.ExtensionManagerAdapter;
+import org.mule.runtime.module.extension.internal.runtime.operation.OperationMessageProcessor;
+import org.mule.runtime.module.extension.internal.runtime.operation.OperationMessageProcessorFactory;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticValueResolver;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * This is the default implementation for a {@link ExtensionsClient}, it uses the {@link ExtensionManagerAdapter}
+ * in the {@link MuleContext} to search for the extension that wants to execute the operation from.
+ * <p>
+ * The concrete execution of the operation is handled by an {@link OperationMessageProcessor} instance.
+ * <p>
+ * This implementation can only execute extensions that were built using the SDK, Smart Connectors operations can't be executed.
+ *
+ * @since 4.0
+ */
+public final class DefaultExtensionsClient implements ExtensionsClient, Initialisable {
+
+  @Inject
+  private MuleContext muleContext;
+
+  private ExtensionManagerAdapter extensionManager;
+
+  private final DefaultPolicyManager policyManager = new DefaultPolicyManager();
```

Comment:
don't. PolicyManager needs to be injected as well. You can't just go and create your own